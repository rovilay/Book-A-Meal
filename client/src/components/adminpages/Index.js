import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import navData from '../../helpers/navData';
import { setNav } from '../../actions/navLinks';


class MenuPage extends Component {
  componentDidMount() {
    this.props.dispatch(setNav(navData.adminNavDefault));
  }
  render() {
    return (
      <div>
        admin Page!
      </div>
    );
  }
}

MenuPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(MenuPage);
