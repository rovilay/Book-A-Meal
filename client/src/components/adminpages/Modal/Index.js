import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

import MenuDetailsTable from '../dashboard/MenuTable/MenuDetails';
import EditMenuTable from '../dashboard/MenuTable/EditMenu';
import OrderDetailsTable from '../orderPage/orderDetails';

Modal.setAppElement('#root');
class ModalComp extends Component {
  componentDidMount() {
    this.props.emptyEditMenu();
  }
  render() {
    const { hideModal, modal } = this.props;
    const {
      isOpen,
      isInfo,
      isEdit,
      isOrderInfo,
      content,
      contentLabel,
    } = modal;

    return (
      <Modal
        isOpen={isOpen}
        contentLabel= {contentLabel}
        overlayClassName="overlay"
        className="modal"
      >
        <div className="closeBtn">
          <button
            onClick={hideModal}
            className="btn-col btn-1"
          >
            close
          </button>
        </div>
        {
          (isInfo && content)
          &&
          (
            <MenuDetailsTable
              title={contentLabel}
              content={content}
              {...this.props}
            />
          )
        }

        {
          (isEdit)
          &&
          (
            <EditMenuTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }

        {
          (isOrderInfo)
          &&
          (
            <OrderDetailsTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }
      </Modal>
    );
  }
}

ModalComp.defaultProps = {
  emptyEditMenu: undefined
};

ModalComp.propTypes = {
  modal: PropTypes.object.isRequired,
  setModal: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
  emptyEditMenu: PropTypes.func,
};

export default ModalComp;
