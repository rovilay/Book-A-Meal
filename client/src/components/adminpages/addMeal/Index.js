import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddMeal extends Component {
  render() {
    return (
      <div>
        addmeal page
      </div>
    );
  }
}

export default AddMeal;
