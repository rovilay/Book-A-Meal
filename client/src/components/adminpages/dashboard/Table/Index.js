import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import tableHead from '../../../../helpers/tableHead';
import TableHead from '../../../common/Table/TableHead';
import MenuTableRow from './MenuTableRow';

class MenuTable extends Component {
  render() {
    const { menus } = this.props;
    return (
      <div className="menu_table">
        <table>
          <TableHead tableHead={tableHead.menuTableHead} />
          <tbody>
            {
              menus.map((menu, i) => {
                const {
                  id: menuId,
                  postOn,
                  User,
                } = menu;
                const item = {
                  sn: ++i,
                  menuId,
                  postOn: moment(postOn).format('LL'),
                  createdBy: `${User.firstName} ${User.lastName}`
                };
                return (<MenuTableRow
                  key={menuId}
                  item={item}
                />);
              })
            }
          </tbody>
        </table>
      </div>
    );
  }
}

MenuTable.propTypes = {
  menus: PropTypes.array.isRequired
};

export default MenuTable;
