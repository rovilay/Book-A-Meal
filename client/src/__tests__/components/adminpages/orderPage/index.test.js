import React from 'react';
import { shallow } from 'enzyme';
import { OrderHistory, mapStateToProps } from '../../../../components/adminpages/orderPage/Index';
import { modalDefaultState } from '../../../../reducers/modalReducer';
import { ordersDefaultState } from '../../../../reducers/ordersReducer';
import { filterDefaultState } from '../../../../reducers/filterReducer';
import { orders } from '../../../__mockData__/orderMock';


const props = {
  pagination: {
    limit: 12,
    count: 13,
    numOfPages: 2
  },
  setModal: jest.fn(),
  modal: {
    isOpen: true,
    isEdit: true,
    isOrderInfo: true,
    contentLabel: 'Modal',
    content: {}
  },
  getAllOrders: jest.fn(),
  setFilter: jest.fn(),
  orders,
  setNav: jest.fn(),
  grandTotalPrice: 1100,
  getOrderMeals: () => Promise.resolve()
};

describe('OrderHistory component test', () => {
  const setup = () => shallow(<OrderHistory {...props} />);


  it('should render `OrderHistory` correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });


  it('should call `handlePaginationClick`', (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(wrapper.instance(), 'handlePaginationClick');
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });


  it('should call `showDetails`', (done) => {
    const wrapper = setup();


    const showDetailsSpy= jest.spyOn(wrapper.instance(), 'showDetails');
    wrapper.instance().showDetails('orders/1/meals')
    expect(showDetailsSpy).toHaveBeenCalled();

    done();
  });


  it('should display `No orders found!` if no orders', (done) => {
    props.orders = [];
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should map state to props', () => {
    const initialState = {
      modal: {
        ...modalDefaultState
      },
      orders: {
        ...ordersDefaultState
      },
      filter: {
        ...filterDefaultState
      }
    };

    const ownProps = {
      orders: props.orders,
      modal: props.modal,
      pagination: props.pagination,
      grandTotalPrice: props.grandTotalPrice,
      orderedMealsPagination: props.pagination
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  })
});
