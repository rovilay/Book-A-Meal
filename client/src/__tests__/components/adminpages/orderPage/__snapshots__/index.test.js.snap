// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistory component test should display \`No orders found!\` if no orders 1`] = `
<div
  className="admin-orderPage"
>
  <div
    className="welcome"
  >
    <p
      className="merienda"
    >
      Order History
    </p>
  </div>
  <div
    className="table-container"
  >
    <FilterComp
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
            Array [
              Object {},
            ],
            Array [
              Object {
                "limit": 12,
                "offset": 12,
              },
            ],
            Array [
              Object {},
            ],
            Array [
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      getOrderMeals={[Function]}
      grandTotalPrice={1100}
      modal={
        Object {
          "content": Object {},
          "contentLabel": "Modal",
          "isEdit": true,
          "isOpen": true,
          "isOrderInfo": true,
        }
      }
      orders={Array []}
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
            Array [
              Object {
                "filter": "all",
              },
            ],
            Array [
              Object {
                "filter": "all",
              },
            ],
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      tableContent="order_history"
    />
  </div>
  <p
    className="empty not-found"
  >
    No orders found!
  </p>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="break-me"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={2}
    nextClassName="next"
    nextLabel=">>"
    onPageChange={[Function]}
    pageCount={2}
    pageRangeDisplayed={5}
    previousClassName="previous"
    previousLabel="<<"
    subContainerClassName="pages pagination"
  />
  <ModalComp
    getAllOrders={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
          Array [
            Object {
              "limit": 12,
              "offset": 12,
            },
          ],
          Array [
            Object {},
          ],
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getOrderMeals={[Function]}
    grandTotalPrice={1100}
    hideModal={[Function]}
    modal={
      Object {
        "content": Object {},
        "contentLabel": "Modal",
        "isEdit": true,
        "isOpen": true,
        "isOrderInfo": true,
      }
    }
    orders={Array []}
    pagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setFilter={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "filter": "all",
            },
          ],
          Array [
            Object {
              "filter": "all",
            },
          ],
          Array [
            Object {
              "filter": "all",
            },
          ],
          Array [
            Object {
              "filter": "all",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setModal={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
              "isOrderInfo": false,
            },
          ],
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
              "isOrderInfo": false,
            },
          ],
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
              "isOrderInfo": false,
            },
          ],
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
              "isOrderInfo": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setNav={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />
</div>
`;

exports[`OrderHistory component test should map state to props 1`] = `
Object {
  "grandTotalPrice": 0,
  "modal": Object {
    "close": true,
    "content": Object {},
    "contentLabel": "",
    "isEdit": false,
    "isInfo": false,
    "isOpen": false,
    "isSetMenu": false,
    "pagination": Object {
      "limit": 10,
      "numOfPages": 1,
      "offset": 0,
    },
  },
  "orderedMealsPagination": Object {
    "count": 0,
    "limit": 5,
    "numOfPages": 1,
    "offset": 0,
  },
  "orders": Array [],
  "pagination": Object {
    "count": 0,
    "limit": 10,
    "numOfPages": 1,
    "offset": 0,
  },
}
`;

exports[`OrderHistory component test should render \`OrderHistory\` correctly 1`] = `
<div
  className="admin-orderPage"
>
  <div
    className="welcome"
  >
    <p
      className="merienda"
    >
      Order History
    </p>
  </div>
  <div
    className="table-container"
  >
    <FilterComp
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      getOrderMeals={[Function]}
      grandTotalPrice={1100}
      modal={
        Object {
          "content": Object {},
          "contentLabel": "Modal",
          "isEdit": true,
          "isOpen": true,
          "isOrderInfo": true,
        }
      }
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      tableContent="order_history"
    />
  </div>
  <div
    className="container-test"
  >
    <div
      className="row head"
    >
      <p
        className="row-item"
        key="0"
      >
        S/N
      </p>
      <p
        className="row-item"
        key="1"
      >
        OrderId
      </p>
      <p
        className="row-item"
        key="2"
      >
        Date
      </p>
      <p
        className="row-item"
        key="3"
      >
        Total Cost
      </p>
      <p
        className="row-item"
        key="4"
      >
        Actions
      </p>
    </div>
    <OrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      getOrderMeals={[Function]}
      grandTotalPrice={1100}
      item={
        Object {
          "date": "August 23, 2018",
          "orderId": 1,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="1"
      mealsUrl="api/v1/orders/1"
      modal={
        Object {
          "content": Object {},
          "contentLabel": "Modal",
          "isEdit": true,
          "isOpen": true,
          "isOrderInfo": true,
        }
      }
      orderCreatedAt="2018-08-23"
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={2}
    />
    <OrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      getOrderMeals={[Function]}
      grandTotalPrice={1100}
      item={
        Object {
          "date": "August 24, 2018",
          "orderId": 2,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="2"
      mealsUrl="api/v1/orders/2"
      modal={
        Object {
          "content": Object {},
          "contentLabel": "Modal",
          "isEdit": true,
          "isOpen": true,
          "isOrderInfo": true,
        }
      }
      orderCreatedAt="2018-08-24"
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={3}
    />
    <OrderTableRow
      actions={
        Object {
          "delete": true,
          "edit": true,
          "info": true,
        }
      }
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      getOrderMeals={[Function]}
      grandTotalPrice={1100}
      item={
        Object {
          "date": "August 25, 2018",
          "orderId": 3,
          "sn": NaN,
          "totalPrice": undefined,
        }
      }
      key="3"
      mealsUrl="api/v1/orders/3"
      modal={
        Object {
          "content": Object {},
          "contentLabel": "Modal",
          "isEdit": true,
          "isOpen": true,
          "isOrderInfo": true,
        }
      }
      orderCreatedAt="2018-08-25"
      orders={
        Array [
          Object {
            "Meals": "api/v1/orders/1",
            "createdAt": "2018-08-23",
            "id": 1,
          },
          Object {
            "Meals": "api/v1/orders/2",
            "createdAt": "2018-08-24",
            "id": 2,
          },
          Object {
            "Meals": "api/v1/orders/3",
            "createdAt": "2018-08-25",
            "id": 3,
          },
        ]
      }
      pagination={
        Object {
          "count": 13,
          "limit": 12,
          "numOfPages": 2,
        }
      }
      setFilter={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "filter": "all",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "close": true,
                "contentLabel": "",
                "isEdit": false,
                "isInfo": false,
                "isOpen": false,
                "isOrderInfo": false,
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      setNav={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "link": "/dashboard",
                  "title": "Manage Menus",
                },
                Object {
                  "link": "/mealpage",
                  "title": "Manage Meals",
                },
                Object {
                  "link": "/orderHistory",
                  "title": "Orders",
                },
                Object {
                  "link": "/",
                  "title": "Log Out",
                },
              ],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      showDetails={[Function]}
      sn={4}
    />
  </div>
  <div
    className="order"
  >
    <p
      className="grandTot"
    >
      Total Sales: ₦
      1100
    </p>
  </div>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="break-me"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={2}
    nextClassName="next"
    nextLabel=">>"
    onPageChange={[Function]}
    pageCount={2}
    pageRangeDisplayed={5}
    previousClassName="previous"
    previousLabel="<<"
    subContainerClassName="pages pagination"
  />
  <ModalComp
    getAllOrders={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getOrderMeals={[Function]}
    grandTotalPrice={1100}
    hideModal={[Function]}
    modal={
      Object {
        "content": Object {},
        "contentLabel": "Modal",
        "isEdit": true,
        "isOpen": true,
        "isOrderInfo": true,
      }
    }
    orders={
      Array [
        Object {
          "Meals": "api/v1/orders/1",
          "createdAt": "2018-08-23",
          "id": 1,
        },
        Object {
          "Meals": "api/v1/orders/2",
          "createdAt": "2018-08-24",
          "id": 2,
        },
        Object {
          "Meals": "api/v1/orders/3",
          "createdAt": "2018-08-25",
          "id": 3,
        },
      ]
    }
    pagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setFilter={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "filter": "all",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setModal={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "close": true,
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
              "isOrderInfo": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setNav={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />
</div>
`;
